jobs/periodic/flag-old-conversations/index.ts

```diff
@@ -55,42 +55,58 @@ export default async function flagInactiveConversationsAsReadonly() {
         const { readMembers } = checkChatMembersAccessRights(conversation);
         const isChatReadOnly = readMembers.length === countParticipants;
         if (isChatReadOnly) {
-            return;
+            logger.info(`Conversation ${conversation.id} is already readonly.`);
         }
 
-        if (conversation.custom.subcourse) {
-            const subcourseIds: number[] = JSON.parse(conversation.custom.subcourse);
-            const allSubcoursesActive = await Promise.all(subcourseIds.map((id) => isActiveSubcourse(id)));
-            shouldMarkAsReadonly = allSubcoursesActive.every((active) => active === false);
-        }
-        if (conversation.custom.prospectSubcourse) {
-            const prospectSubcourses: number[] = conversation.custom.prospectSubcourse;
-            const allProspectSubcoursesActive = await Promise.all(prospectSubcourses.map((id) => isActiveSubcourse(id)));
-            shouldMarkAsReadonly = allProspectSubcoursesActive.every((active) => active === false);
-        }
+        if (!isChatReadOnly) {
```

Comment:
I'd prefer the short curcuit with return instead of an indirect if else

Reply:
However, this has prevented further conversations from being checked once a conversation is already readonly. Do you know why?

Reply:
Ah true, could use `continue` instead of `return`, or an if/else